src/test/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExampleTest.java

```diff
@@ -139,102 +146,281 @@ public void shouldDemonstrateInteractiveQueries() throws Exception {
+                      .request(MediaType.APPLICATION_JSON_TYPE).get(HostStoreInfo.class);
+    
+      // Fetch the value for the key hello from the instance.
+      final KeyValueBean result = client.target("http://" + hostWithHelloKey.getHost() +
+              ":" + hostWithHelloKey.getPort() +
+              "/state/keyvalue/word-count/hello")
+              .request(MediaType.APPLICATION_JSON_TYPE).get(KeyValueBean.class);
+    
+      assertThat(result, equalTo(new KeyValueBean("hello", 2L)));
+    
+      // fetch windowed values for a key
+      final List<KeyValueBean>
+              windowedResult =
+              client.target(baseUrl + "/windowed/windowed-word-count/streams/0/" + System
+                      .currentTimeMillis())
+                      .request(MediaType.APPLICATION_JSON_TYPE)
+                    
+                      .get(new GenericType<List<KeyValueBean>>() {
+                      });
+      assertThat(windowedResult.size(), equalTo(1));
+      final KeyValueBean keyValueBean = windowedResult.get(0);
+      assertTrue(keyValueBean.getKey().startsWith("streams"));
+      assertThat(keyValueBean.getValue(), equalTo(3L));
+    }
+  }
+  
+  @Test(expected = Exception.class)
+  public void shouldThrowExceptionForInvalidHost() throws Exception {
+    final int port = randomFreeLocalPort();
+    final String host = "someInvalidHost";
+    
+    kafkaStreams = WordCountInteractiveQueriesExample.createStreams(
+            createStreamConfig(CLUSTER.bootstrapServers(), port, "one", host));
+    
+    final CountDownLatch startupLatch = new CountDownLatch(1);
+    kafkaStreams.setStateListener((newState, oldState) -> {
+      if (newState == KafkaStreams.State.RUNNING && oldState == KafkaStreams.State.REBALANCING) {
+        startupLatch.countDown();
+      }
+    });
+    
+    kafkaStreams.start();
+    proxy = WordCountInteractiveQueriesExample.startRestProxy(kafkaStreams, port, host);
+  }
+  
+  @Test
+  public void shouldThrowBindExceptionForUnavailablePort() throws Exception {
+    expectedEx.expect(Exception.class);
+    expectedEx.expectMessage("java.net.BindException: Address already in use");
+    
+    final int port = randomFreeLocalPort();
+    final String host = "localhost";
+    
+    kafkaStreams = WordCountInteractiveQueriesExample.createStreams(
+            createStreamConfig(CLUSTER.bootstrapServers(), port, "one", host));
+    
+    final CountDownLatch startupLatch = new CountDownLatch(1);
+    kafkaStreams.setStateListener((newState, oldState) -> {
+      if (newState == KafkaStreams.State.RUNNING && oldState == KafkaStreams.State.REBALANCING) {
+        startupLatch.countDown();
+      }
+    });
+    
+    kafkaStreams.start();
+    proxy = WordCountInteractiveQueriesExample.startRestProxy(kafkaStreams, port, host);
+    // Binding to same port again will raise BindException.
+    WordCountInteractiveQueriesExample.startRestProxy(kafkaStreams, port, host);
```

Comment:
We should update this to:
```
try {
    WordCountInteractiveQueriesExample.startRestProxy(kafkaStreams, port, host);
    fail("Should have throw BindException"):
} catch (final BindException expected) {}
```

Otherwise, if a `BindException` is throw before this statement, the test would pass.